runbotics/runbotics-orchestrator-ui/src/src-app/store/slices/ProcessInstance/ProcessInstance.utils.ts

```diff
@@ -1,27 +1,63 @@
 import { ProcessInstanceState, InstanceExtendedWithSubprocesses } from './ProcessInstance.state';
 
-const recursivelyInsertSubprocess = (
-    parentInstanceId: string, 
-    currentNode: InstanceExtendedWithSubprocesses, 
-    targetSubprocesses: InstanceExtendedWithSubprocesses[],
-) => {
+const spreadIfArray = (value: Array<unknown> | unknown) => Array.isArray(value) ? value : [];
```

Comment:
I find this name a little bit misleading - it suggest the spread is already done here and I see you use it to check if it needs speading. maybe you can return already spreaded value here? and when you use it you could do this [spreadIfArray(currentNode.subprocesses), spreadIfArray(targetSubprocesses)]?